core/variant/variant_construct.cpp

```diff
@@ -39,6 +39,59 @@
 #include "core/templates/local_vector.h"
 #include "core/templates/oa_hash_map.h"
 
+template <class T>
+struct PtrConstruct {};
+
+#define MAKE_PTRCONSTRUCT(m_type)                                                  \
+	template <>                                                                    \
+	struct PtrConstruct<m_type> {                                                  \
+		_FORCE_INLINE_ static void construct(const m_type &p_value, void *p_ptr) { \
+			memnew_placement(p_ptr, m_type(p_value));                              \
+		}                                                                          \
+	};
+
+template <>
+struct PtrConstruct<Object *> {
+	_FORCE_INLINE_ static void construct(Object *p_value, void *p_ptr) {
+		*((Object **)p_ptr) = p_value;
+	}
+};
+
+MAKE_PTRCONSTRUCT(bool);
+MAKE_PTRCONSTRUCT(int64_t);
+MAKE_PTRCONSTRUCT(double);
+MAKE_PTRCONSTRUCT(String);
+MAKE_PTRCONSTRUCT(Vector2);
+MAKE_PTRCONSTRUCT(Rect2);
+MAKE_PTRCONSTRUCT(Vector3);
+MAKE_PTRCONSTRUCT(Vector2i);
+MAKE_PTRCONSTRUCT(Rect2i);
+MAKE_PTRCONSTRUCT(Vector3i);
```

Comment:
I suggest using the same order as in the `Variant::Type` enum, that makes it easier to parse the list and make sure nothing is missing.

Reply:
TBH I just copied the list from `method_ptrcall.h` but makes sense to keep the order.

Reply:
Might be worth changing in `method_ptrcall.h` at the same time then.

Reply:
Okay, I added a commit to change the other file too.